### ISSUE TITLE
armeria-jetty: ArmeriaConnector - NPE when DEBUG logging enabled

### ISSUE THREAD

**Issue by max904-github** [line/armeria#2688]
### Problem:
To reproduce, simply enable `DEBUG` logging via Logback and run the unit tests for armeria-jetty project...

You'll see numerous exceptions like below:
```
SLF4J: Failed toString() invocation on an object of type [com.linecorp.armeria.server.jetty.ArmeriaConnector]
Reported exception:
java.lang.NullPointerException
	at com.linecorp.armeria.server.jetty.ArmeriaConnector.getHost(ArmeriaConnector.java:64)
	at org.eclipse.jetty.server.AbstractNetworkConnector.toString(AbstractNetworkConnector.java:122)
	at org.slf4j.helpers.MessageFormatter.safeObjectAppend(MessageFormatter.java:277)
	at org.slf4j.helpers.MessageFormatter.deeplyAppendParameter(MessageFormatter.java:249)
	at org.slf4j.helpers.MessageFormatter.arrayFormat(MessageFormatter.java:211)
	at org.slf4j.helpers.MessageFormatter.arrayFormat(MessageFormatter.java:161)
	at org.eclipse.jetty.util.log.JettyAwareLogger.log(JettyAwareLogger.java:624)
	at org.eclipse.jetty.util.log.JettyAwareLogger.debug(JettyAwareLogger.java:206)
	at org.eclipse.jetty.util.log.Slf4jLog.debug(Slf4jLog.java:95)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.addBean(ContainerLifeCycle.java:412)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.addBean(ContainerLifeCycle.java:330)
	at org.eclipse.jetty.server.AbstractConnector.<init>(AbstractConnector.java:211)
	at org.eclipse.jetty.server.AbstractNetworkConnector.<init>(AbstractNetworkConnector.java:44)
	at org.eclipse.jetty.server.ServerConnector.<init>(ServerConnector.java:212)
	at org.eclipse.jetty.server.ServerConnector.<init>(ServerConnector.java:129)
	at com.linecorp.armeria.server.jetty.ArmeriaConnector.<init>(ArmeriaConnector.java:46)
	at com.linecorp.armeria.server.jetty.JettyService.start(JettyService.java:160)
	at com.linecorp.armeria.server.jetty.JettyService$Configurator.serverStarting(JettyService.java:463)
	at com.linecorp.armeria.server.Server$ServerStartStopSupport.notifyStarting(Server.java:630)
	at com.linecorp.armeria.server.Server$ServerStartStopSupport.notifyStarting(Server.java:377)
	at com.linecorp.armeria.common.util.StartStopSupport.notifyListeners(StartStopSupport.java:371)
	at com.linecorp.armeria.common.util.StartStopSupport.lambda$start0$3(StartStopSupport.java:184)
	at io.netty.util.concurrent.GlobalEventExecutor$TaskRunner.run(GlobalEventExecutor.java:243)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:830)

```
### Analysis:
`ArmeriaConnector` extends Jetty's `ServerConnector`. Here is its simplified constructor definition (note that `armeriaServer` only assigned after `super` constructor called):
```
    ArmeriaConnector(Server server, com.linecorp.armeria.server.Server armeriaServer) {
        super(server, -1, -1, new ArmeriaConnectionFactory());
        this.armeriaServer = armeriaServer;
        ...
    }
```
While `getHost()`, `getPort()`, `getLocalPort()`, etc. methods overridden like following:
```
    @Override
    public String getHost() {
        return armeriaServer.defaultHostname();
    }
```
The problem occurs in `DEBUG` mode, where Jetty's `ServerConnector` internally logs its state during construction, which assumes invocation of `getHost()` and `getPort()` methods, which will cause NPE.
### Possible solution:
`ArmeriaConnector` methods like `getHost()` and `getPort()` could check `armeriaServer` state and avoid `NullPointerException` during components initialization.

**Comment by trustin** [comment]
Thanks a lot for the detailed analysis. Your suggested solution sounds good to me.

**Comment by max904-github** [comment]
Since you agreed with the proposed solution, I can make this fix and test it.

**Comment by trustin** [comment]
Awesome! Thanks a lot in advance. :bow:

**Comment by max904-github** [comment]
Pushed the fix.

**Comment by trustin** [comment]
Fixed via #2691. Thanks again, @max904-github
